src/web/js/check-ui.js

```diff
@@ -297,220 +477,90 @@
+        }, function(result) {
+          if (runtime.isSuccessResult(result)) {
+            runtime.runThunk(function() {
+              return outputUI.renderErrorDisplay(documents, runtime, result.result, []);
+            }, function(result) {
+              if (runtime.isSuccessResult(result)) {
+                if(skeleton.pyretStack)
+                  result.result.append(outputUI.renderStackTrace(runtime, documents, srcloc, skeleton.pyretStack));
+                skeleton.vivify(result.result);
+                restarter.resume(runtime.nothing);
               } else {
-                var snippet =
-                  outputUI.snippet(editors,
-                                   outputUI.cmPosFromSrcloc(runtime, srcloc, get(test,"loc")))
-                  .wrapper;
-                reason.replaceWith(snippet);
-                block.one("click", ".check-block-header",
-                          function(e) {
-                            snippet[0].cmrefresh();
-                          });
+                skeleton.vivify(
+                  $("<span>").text("An error occurred rendering the reason for this error; details logged to the console"));
+                console.error(result);
+                restarter.resume(runtime.nothing);
               }
-            }), 0, tests.length);
-          }), 0, checkBlockTestReports.length)
-        }, function(ans){
-          outputUI.runMarks();
-          resultsContainer.append($("<span class='check-results-done-rendering' style='display:none'></span>"));
-          return ans;
+            });
+          } else {
+            skeleton.vivify(
+              $("<span>").text("An error occurred rendering the reason for this error; details logged to the console"));
+            console.error(result);
+            restarter.resume(runtime.nothing);
+          }
+        });
+      }
+      
+      function vivifySkeleton(skeleton) {
+        runtime.pauseStack(function(restarter){
+          runtime.runThunk(function(){
+            return get(skeleton.renderable, "render-fancy-reason")
+                    .app((skeleton.hasOwnProperty("maybeStackLoc") ? 
+                            skeleton.maybeStackLoc : maybeStackLoc), 
+                          srclocAvaliable, maybeLocToAST);
+          },
+          function(result) {
+            if (runtime.isSuccessResult(result)) {
+              runtime.runThunk(
+                function(){
+                  return outputUI.renderErrorDisplay(documents, runtime, result.result, []);
+                }, function(result) {
+                  if (runtime.isSuccessResult(result)) {
+                    if(skeleton.pyretStack)
+                      result.result.append(outputUI.renderStackTrace(runtime, documents, srcloc, skeleton.pyretStack));
+                    skeleton.vivify(result.result);
+                    restarter.resume(runtime.nothing);
+                  } else vivifySkeletonFallback(restarter, skeleton);
```

Comment:
Should there be some error message in the console here, that would indicate something went wrong?  This is an utterly silent failure of renderErrorDisplay; likewise the one below is utterly silent about render-fancy-reason...